internal/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,277 @@
+		mountNsMapSetter: mountNsMapSetter,
+		containers:       gadgetContext.Context().Value(TestOperatorContext("containers")).([]*containercollection.Container),
+		testingTB:        gadgetContext.Context().Value(TestOperatorContext("testing.TB")).(testing.TB),
+	}, nil
+}
+
+type TestOperatorInstance struct {
+	*TestOperator
+	attacher         Attacher
+	mountNsMapSetter MountNsMapSetter
+	containers       []*containercollection.Container
+	testingTB        testing.TB
+	tc               *tracercollection.TracerCollection
+}
+
+func (t *TestOperatorInstance) PreGadgetRun() error {
+	if t.attacher != nil {
+		for i := range t.containers {
+			err := t.attacher.AttachContainer(t.containers[i])
+			if err != nil {
+				t.testingTB.Fatal(err)
+			}
+		}
+	}
+	if t.mountNsMapSetter != nil {
+		tc, err := tracercollection.NewTracerCollectionTest(nil)
+		if err != nil {
+			t.testingTB.Fatal(err)
+		}
+		t.tc = tc
+		tc.AddTracer("test-tracer", containercollection.ContainerSelector{})
+
+		mountnsmap, err := tc.TracerMountNsMap("test-tracer")
+		if err != nil {
+			t.testingTB.Fatal(err)
+		}
+		t.mountNsMapSetter.SetMountNsMap(mountnsmap)
+	}
+	return nil
+}
+
+func (t *TestOperatorInstance) PostGadgetRun() error {
+	if t.attacher != nil {
+		for i := range t.containers {
+			err := t.attacher.DetachContainer(t.containers[i])
+			if err != nil {
+				t.testingTB.Fatal(err)
+			}
+
+		}
+	}
+	if t.mountNsMapSetter != nil {
+		t.tc.Close()
+	}
+	return nil
+}
+
+func (t *TestOperatorInstance) EnrichEvent(ev any) error {
+	return nil
+}
+
+func init() {
+	operators.Register(&TestOperatorInstance{})
+}
+
+func BenchmarkAllGadgetsWithContainers(b *testing.B) {
+	utilstest.RequireRoot(b)
+
+	// Prepare runtime
+	runtime := &local.Runtime{}
+	err := runtime.Init(nil)
+	if err != nil {
+		b.Fatalf("initializing runtime: %s", err)
+	}
+	defer runtime.Close()
+
+	containerCounts := []int{0, 1, 10, 100}
+	for _, containerCount := range containerCounts {
+		b.Run(fmt.Sprintf("container%d", containerCount), func(b *testing.B) {
+			// Prepare fake containers
+			runnerConfig := &utilstest.RunnerConfig{}
+			var containers []*containercollection.Container
+			for i := 0; i < containerCount; i++ {
+				runner := utilstest.NewRunnerWithTest(b, runnerConfig)
+				container := &containercollection.Container{
+					ID:    fmt.Sprintf("container%d", i),
+					Mntns: runner.Info.MountNsID,
+					Netns: runner.Info.NetworkNsID,
+					Pid:   uint32(runner.Info.Tid),
+				}
+				containers = append(containers, container)
+			}
+
+			allGadgets := gadgetregistry.GetAll()
+
+			for _, gadgetDesc := range allGadgets {
+				gadgetDesc := gadgetDesc
+
+				// Skip unwanted gadgets
+				categoryAndName := fmt.Sprintf("%s-%s", gadgetDesc.Category(), gadgetDesc.Name())
+				skipRegex := os.Getenv("IG_BENCHMARKS_GADGET_REGEX")
+				if skipRegex != "" {
+					matched, _ := regexp.Match(skipRegex, []byte(categoryAndName))
+					if !matched {
+						continue
+					}
+				}
```

Comment:
Good point. I didn't know how to make the regex work. It seems it is a list of `/`-separated regex: I have to use `^` and `$` on each section of the path to really select what I want, otherwise `container10` will also capture `container100`:
```
$ go test -exec sudo -bench='^BenchmarkAllGadgetsWithContainers$/^container10$/trace-tcpconnect' -run=Benchmark ./internal/benchmarks/...
```
I will add an example in the Makefile and remove the custom Golang regex code.